### PR TITLE
Implement context aware validation functions

### PR DESCRIPTION
**Pull request by egor-denysenko**
## Description

This patch implements support for existing markers for context-aware validation functions. With the goal of honour context before each struct attribute validation.

With this patch also middleware utility was adapted to honor the http request context during validation. 

Documentation was also updated by adding an additional page for describing context-aware functions usage.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update
- [ ] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [X] Code passes `make golangci-lint` without errors
- [X] All tests pass: `make test`
- [X] Code follows existing patterns and conventions
- [X] No hard-coded values (use constants or configuration)
- [X] Error messages are clear and actionable

### Testing Requirements
- [X] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [X] Unit tests implemented with boundary value testing (`test/unit/`)
- **_Maybe it could be done only for go-playground/validator since it's also support context methods. Up for discussion._**
- [ ] ~~Benchmark tests comparing against popular validation libraries (`test/benchmark/`)~~
  - [ ] ~~[[go-playground/validator](https://github.com/go-playground/validator)](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)~~
  - [ ] ~~[[asaskevich/govalidator](https://github.com/asaskevich/govalidator)](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)~~
  - [ ] ~~[[gookit/validate](https://github.com/gookit/validate)](https://github.com/gookit/validate) (BenchmarkGookitValidate*)~~
- [X] All test files generated: `cd test && go generate`

### Performance Standards (not applicable)
- [ ] ~~Zero allocations in validation logic (verified via benchmarks)~~
- [ ] ~~Performance improvement over existing validation libraries (minimum 2x faster)~~
- [ ] ~~Benchmark results added to `test/benchmark/README.md`~~

### Documentation
- [X] README.md updated with new marker documentation
- [X] Code includes appropriate comments and examples

### Final Verification
- [X] All GitHub Actions CI checks pass
- [X] No breaking changes to existing functionality
- [X] Contribution follows project license requirements

## Additional Notes
Some thoughts that could be addressed:
- Bumping golangci-lint to a more recent version v2.9.0 has been released and adds support for 1.26 (currently linter sits at v2.1.6
- Same for lefthook v2 is available (github.com/evilmartians/lefthook/v2) and tool can be migrated since v1 is used currently.
- Lib could be bumped to 1.25 since 1.26 has been released and 1.24 is not maintained anymore (providing additional packages like synctest for improving tests written with this patch)
- Should go version in readme resemble the one in go.mod https://github.com/sivchari/govalid/blob/main/README.md?plain=1#L6? 
```
~
❯ go version
go version go1.21.0 linux/amd64

~
❯ go install github.com/sivchari/govalid/cmd/govalid@latest
go: downloading github.com/sivchari/govalid v1.8.0
go: github.com/sivchari/govalid/cmd/govalid@latest: github.com/sivchari/govalid@v1.8.0 requires go >= 1.24.3 (running go 1.21.0; GOTOOLCHAIN=go1.21.0)
```

## Related Issues

Implements #87

### CODE REVIEW

File: internal/analyzers/govalid/templates/validation.go.tmpl [hunk 1]
```diff
@@ -40,7 +44,17 @@ func Validate{{.TypeName}}(t *{{.TypeName}}) error {
 
 		{{ if and (ne $parentVariable "") ( .Validators ) -}}
 	    	{
-  			t := t.{{ $parentVariable }}
+				if ctx.Err() != nil {
+					return ctx.Err()
+				}
```
Comment (sivchari) [inline internal/analyzers/govalid/templates/validation.go.tmpl]: I think it's enough to check just once first.

**Comment by sivchari** [comment]
Great job! Thx!
